utils/gyb.py

```diff
@@ -716,7 +716,8 @@ def execute(self, context):
 
         # If we got a result, the code was an expression, so append
         # its value
-        if result is not None and result != '':
+        if result is not None \
+                or (isinstance(result, basestring) and result != ''):
```

Comment:
Out of curiousity I just ran `gyb.py --verbose-test` with both python and python3. Both succeeded. Which is a bit surprising since doctests contain a lot of `print x` statements. Anyhow, [this](http://stackoverflow.com/a/22679982/79386) looks alright.